Salesforce Examples Topic - DOCS-11025 (#1486)
mulesoft/docs-connectors: salesforce/10.11/modules/ROOT/pages/salesforce-example-parent-child.adoc

== Create or Update an Object in a Parent-Child Relationship

A Salesforce object can have standard or custom relationships between objects. The relationships between objects are usually one-to-many (parent-child) relationships, but can be any link between two objects residing in Salesforce.

This example shows how to create the object relationship structure in Salesforce needed to perform an upsert using Salesforce Connector. The example uses two custom Salesforce object types that have a parent-child relationship:

* `Plane__c`, the parent object that must exist in Salesforce

> * `Component__c`, the object to upsert using Salsforce Connector

Reviewer: ```suggestion
* `Component__c`, the object to upsert using Salesforce Connector
```

The relationship between the two custom objects is represented by a `Plane__r` field whose value is an object with two fields:

* The `type` field, which has the referenced object type name as its value.
+

> In this example, the value of `type` is `Plane_c`.

Reviewer: ```suggestion
In this example, the value of `type` is `Plane_c`, the parent object.
```

> The relationship between the two custom objects is represented by a `Plane__r` field whose value is an object with two fields:
> 
> * The `type` field, which has the referenced object type name as its value.
> +
> In this example, the value of `type` is `Plane_c`.
> +
> * A field that identifies the instance of the parent object to reference.
> +
> In this example, the `Name` field references the `Cobra` instance of `Plane__c`.

Reviewer: ```suggestion
The relationship between the two custom objects is represented by another object named `Plane__r`, whose value has two key-value pair fields:

* The `type` key references the parent object type name as its value.
+
In this example, the value of `type` is `Plane_c`.
+
* The `Name` key gives an identity to the instance of the parent object to reference.
+
In this example, the value of `Name` is `Cobra`.
```

Reviewer: Because I found it confusing the previous explanation, I added some rewording suggestions that could explain what I understood from reading the code.
Please reconfirm with the devs to make sure.
